app/src/main/java/com/example/plantsapp/presentation/ui/tasksfordays/TasksForDaysFragment.kt

```diff
@@ -26,20 +31,45 @@ class TasksForDaysFragment : Fragment(R.layout.fragment_tasks_for_days) {
                     requireActivity(),
                     it
                 )
-            }
 
-            currentPageDate.observe(viewLifecycleOwner) {
-                binding.tvCurrentDate.text = it.toString()
+                binding.vpTasks.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+                    override fun onPageSelected(position: Int) {
+                        super.onPageSelected(position)
+
+                        binding.btnPrevDate.text = requireContext().getNewPageText(it, position - 1)
+                        binding.btnNextDate.text = requireContext().getNewPageText(it, position + 1)
+                        binding.tvCurrentDate.text = requireContext().getNewPageText(it, position, true)
+
+                        binding.btnPrevDate.isVisible = isPrevButtonVisible(position)
+                    }
+                })
             }
         }
 
         with(binding) {
-            vpTasks.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
-                override fun onPageSelected(position: Int) {
-                    super.onPageSelected(position)
-                    viewModel.onCurrentPageChange(position)
+            btnPrevDate.setOnClickListener { vpTasks.currentItem -= 1 }
+            btnNextDate.setOnClickListener { vpTasks.currentItem += 1 }
+        }
+    }
+
+    private fun Context.getNewPageText(date: Date, offset: Int, showWithYear: Boolean = false): String {
+        return when (offset) {
+            0 -> resources.getString(R.string.title_today_date)
```

Comment:
1. You can just use getString, without resources
2. Move date formats to private constant in companion object